src/simulation/simulation_state.jl

```diff
@@ -1,32 +1,61 @@
 struct StateInfo
-    aux_variable_values::Dict{AuxVarKey, DataFrames.DataFrame}
-    variable_values::Dict{VariableKey, DataFrames.DataFrame}
-    dual_values::Dict{ConstraintKey, DataFrames.DataFrame}
-    parameter_values::Dict{ParameterKey, DataFrames.DataFrame}
-    expression_values::Dict{ExpressionKey, DataFrames.DataFrame}
+    aux_variables::Dict{AuxVarKey, JuMP.Containers.DenseAxisArray{Float64}}
+    variables::Dict{VariableKey, JuMP.Containers.DenseAxisArray{Float64}}
+end
+
+function StateInfo()
+    return StateInfo(
+        Dict{AuxVarKey, JuMP.Containers.DenseAxisArray{Float64}}(),
+        Dict{VariableKey, JuMP.Containers.DenseAxisArray{Float64}}(),
+    )
 end
 
 struct SimulationState
     decision_states::StateInfo
     system_state::StateInfo
 end
 
-#================================================================
-function update_cache!(
-    sim::Simulation,
-    ::CacheKey{StoredEnergy, D},
-    model::DecisionModel,
-) where {D <: PSY.Device}
-    c = get_cache(sim, StoredEnergy, D)
-    variable = get_variable(model.internal.container, c.ref)
-    t = get_end_of_interval_step(model)
-    for name in variable.axes[1]
-        device_energy = JuMP.value(variable[name, t])
-        @debug name, device_energy
-        c.value[name] = device_energy
-        @debug("Cache value StoredEnergy for device $name set to $(c.value[name])")
+function SimulationState()
+    return SimulationState(StateInfo(), StateInfo())
+end
+
+get_decision_states(s::SimulationState) = s.decision_states
+get_system_state(s::SimulationState) = s.system_state
+
+function _initialize_model_states!(
+    states::StateInfo,
+    model::OperationModel,
+    simulation_step::Dates.Period,
+)
+    container = get_optimization_container(model)
+    model_resolution = get_resolution(model)
+    value_counts = Int(simulation_step / model_resolution)
```

Comment:
Couldn't this produce an InexactError? Use `÷` instead?

Reply:
In theory, this possible error should have been caught earlier. The issue is that if it isn't an integer doing `1:value_counts` will produce Floats and be inconsistent with the indexing.

Reply:
`÷` will always produce an Int. Benefits of this:
- Concise code
- Future readers will never have to wonder whether there is a bug.

I haven't always done this because I didn't realize Julia offered it.

Reply:
I had no idea, fixed via 84ad95d